### PR TITLE
[Enhancement] Get partitions by batch of size 5k per RPC

### PR DESCRIPTION
**Pull request by letian-jiang**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

* Get partitions by batch of size 5k per RPC
* Set max_hive_partitions_per_rpc = 5000

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetaClient.java [hunk 1]
```diff
@@ -228,7 +229,12 @@ public List<Partition> getPartitionsByNames(String dbName, String tblName, List<
             StarRocksConnectorException connectionException = null;
             try {
                 client = getClient();
-                partitions = client.hiveClient.getPartitionsByNames(dbName, tblName, partitionNames);
+                for (int start = 0; start < partitionNames.size(); start += Config.max_hive_partitions_per_rpc) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetaClient.java:232]: you could split partitions in HiveMetastore.java

Comment (letian-jiang) [reply fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetaClient.java]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 8 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastore.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|

**Comment by letian-jiang** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 2]
```diff
@@ -1618,7 +1618,7 @@ public class Config extends ConfigBase {
      * The maximum number of partitions to fetch from the metastore in one RPC.
      */
     @ConfField
-    public static int max_hive_partitions_per_rpc = 1000;
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1621]: Is this variable used before?

Comment (letian-jiang) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java:1621]: Yes. See `CachingHiveMetastore.refreshPartitions`

Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java:1621]: Why increase this value?

Comment (letian-jiang) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java:1621]: Empirical value, suggested by @stephen-shelby

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by letian-jiang** [comment]
@Mergifyio backport branch-3.0

**Comment by letian-jiang** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**